From 6f093297641e07006351a56bf828cd18ebc7d56a Mon Sep 17 00:00:00 2001
From: evilwombat <evilwombat@server.fake>
Date: Sun, 9 Mar 2014 20:34:27 -0700
Subject: [PATCH 12/13] ARM: Ambarella: Always initialize GPIO banks

Since our boot chain is a bit funny (to say the least),
initialize all the GPIO banks regardless of whether the
kernel has been compiled with CONFIG_AMBARELLA_RAW_BOOT
enabled.

Signed-off-by: evilwombat <evilwombat@server.fake>
---
 arch/arm/plat-ambarella/generic/gpio.c |    2 --
 1 file changed, 2 deletions(-)

diff --git a/arch/arm/plat-ambarella/generic/gpio.c b/arch/arm/plat-ambarella/generic/gpio.c
index 09d70ca..e3178f4 100644
--- a/arch/arm/plat-ambarella/generic/gpio.c
+++ b/arch/arm/plat-ambarella/generic/gpio.c
@@ -497,7 +497,6 @@ int __init ambarella_init_gpio(void)
 	}
 #endif
 
-#if defined(CONFIG_AMBARELLA_RAW_BOOT)
 	amba_writel(GPIO0_AFSEL_REG, 0xFFFFFFFF);
 	amba_writel(GPIO0_DIR_REG, 0x00000000);
 	amba_writel(GPIO0_MASK_REG, 0xFFFFFFFF);
@@ -538,7 +537,6 @@ int __init ambarella_init_gpio(void)
 	amba_writel(GPIO5_DATA_REG, 0x00000000);
 	amba_writel(GPIO5_ENABLE_REG, 0xFFFFFFFF);
 #endif
-#endif
 
 	return retval;
 }
-- 
1.7.9.5

